eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/Abstraction/HttpRequestProvider.cs

```diff
@@ -26,15 +27,79 @@ public override ValueExpression Content()
         public override HttpRequestApi FromExpression(ValueExpression original)
             => new HttpRequestProvider(original);
 
-        public override InvokeMethodExpression SetHeaders(IReadOnlyList<ValueExpression> arguments)
-            => Original.Property(nameof(PipelineRequest.Headers)).Invoke(nameof(RequestHeaders.Add), arguments);
+        public override MethodBodyStatement SetHeaders(IReadOnlyList<ValueExpression> arguments)
+            => Original.Property(nameof(PipelineRequest.Headers)).Invoke(nameof(RequestHeaders.SetValue), arguments).Terminate();
 
-        public override AssignmentExpression SetMethod(string httpMethod)
-            => Original.Property(nameof(PipelineRequest.Method)).Assign(New.Instance(typeof(RequestMethod), [Literal(httpMethod)]));
+        public override MethodBodyStatement SetMethod(string httpMethod)
+            => Original.Property(nameof(PipelineRequest.Method)).Assign(CreateRequestMethod(httpMethod)).Terminate();
 
-        public override AssignmentExpression SetUri(ValueExpression value)
-            => Original.Property("Uri").Assign(value);
+        public override MethodBodyStatement SetUri(ValueExpression value)
+            => Original.Property("Uri").Assign(value).Terminate();
 
         public override HttpRequestApi ToExpression() => this;
+
+        private ValueExpression CreateRequestMethod(string httpMethod)
+        {
+            var httpMethodString = ParseHttpMethodString(httpMethod);
+            return httpMethodString is null
+                ? New.Instance(typeof(RequestMethod), [Literal(httpMethod)])
+                : Static<RequestMethod>().Property(httpMethodString);
+        }
+
+        private string? ParseHttpMethodString(string method)
+        {
+            if (method == null)
+            {
+                throw new ArgumentNullException(nameof(method));
+            }
+            if (method.Length == 3)
```

Comment:
I think we could just simply use a `switch` statement here and the compiler could do all these optimizations for us.

Reply:
Actually, taking this logic from [Azure.Core.RequestMethod.Parse](https://github.com/Azure/azure-sdk-for-net/blob/63e086edac99b88fcbc801d1bce49191556b1c8d/sdk/core/Azure.Core/src/RequestMethod.cs#L67-L94)